frontend/lib/editor/incomplete_upload.ts

```diff
@@ -0,0 +1,75 @@
+import { type IncompleteUpload } from '@/openapi/generated/pacta'
+import { Validation, type EditorFieldsFor, type EditorComputedValues } from './common'
+import { getEditorComputedValues } from './utils'
+
+export type EditorIncompleteUpload = EditorFieldsFor<IncompleteUpload>
+
+const createEditorIncompleteUpload = (incompleteUpload: IncompleteUpload): EditorIncompleteUpload => {
+  return {
+    id: {
+      name: 'id',
+      label: 'ID',
+      originalValue: incompleteUpload.id,
+      currentValue: incompleteUpload.id,
```

Comment:
I thought we had killed all this boilerplate with a helper somewhere? I can understand the need for `name` and `label`, but having:

```
originalValue: blah.field,
currentValue: blah.field,
```

seems error-prone and noisy/verbose. Since you're already specifying the keys (e.g. `id`, `name`, etc), you could iterate over them like:

```
const thing = {
	id: {...},
	name: {...},
	description: {...},
}

// This part would go in some helper
for (const key in thing) {
	thing.originalValue = incompleteUpload[key]
	thing.currentValue = incompleteUpload[key]
}
```

Reply:
As soon as this PR is submitted I'm going to work on #57, which I think should provide the best opportunity for making this possible - I'll try to address this concern in that PR.